staging/src/k8s.io/client-go/features/kube_features.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"sync/atomic"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	"k8s.io/component-base/featuregate"
+)
+
+const (
+	// Every feature gate should add method here following this template:
+	//
+	// // owner: @username
+	// // alpha: v1.4
+	// MyFeature featuregate.Feature = "MyFeature"
+	//
+	// Feature gates should be listed in alphabetical, case-sensitive
+	// (upper before any lower case character) order. This reduces the risk
+	// of code conflicts because changes are more likely to be scattered
+	// across the file.
+
+	// owner: @p0lyn0mial
+	// alpha: v1.27
+	//
+	// Allow the API server to stream individual items instead of chunking
+	WatchList featuregate.Feature = "WatchList"
+)
+
+func DefaultFeatureGates() featuregate.FeatureGate {
+	return featureGates.Load().(featuregate.FeatureGate)
+}
+
+func init() {
+	envVarGates := featuregate.NewEnvVarFeatureGate()
+	runtime.Must(envVarGates.Add(defaultKubernetesFeatureGates))
+
+	featureGates.Store(envVarGates)
+}
+
+func AddFeaturesToExistingFeatureGates(mutableFeatureGates featuregate.MutableFeatureGate) error {
+	return mutableFeatureGates.Add(defaultKubernetesFeatureGates)
+}
+
+func SetFeatureGates(newFeatureGates featuregate.FeatureGate) {
+	featureGates.Store(newFeatureGates)
```

Comment:
We also need a better strategy for replacing FG because `Store` will panic when incompatible types are presisted.